src/manager.cpp

```diff
@@ -93,6 +93,121 @@ uint32_t SessionManager::closeAllByType(SessionType type)
     return count;
 }
 
+uint32_t SessionManager::closeUserSessionsByType(SessionType type,
+                                                 std::string sessionId,
+                                                 bool ownedOnly)
+{
+    auto numSessId = parseSessionId(sessionId);
+    std::string userName = getSessionItem(sessionId)->getOwner();
+    bool allowed = isAllSessionsAllowed(sessionId);
+    const auto count =
+        std::erase_if(sessionItems, [allowed, ownedOnly, type, userName,
+                                     numSessId](const auto& item) {
+            auto const [id, session] = item;
+            if ((allowed) && !(ownedOnly))
+            {
+                return session->sessionType() == type && id != numSessId;
+            }
+            else
+            {
+                return session->sessionType() == type &&
+                       session->getOwner() == userName && id != numSessId;
+            }
+        });
+    return count;
+}
+
+bool SessionManager::isAllSessionsAllowed(std::string sessionId)
+{
+    log<level::DEBUG>("SessionManager::isOwnSession()",
+                      entry("CALLER-SESSION-ID=%d", sessionId.c_str()));
+
+    std::map<std::string,
+             std::variant<std::string, std::vector<std::string>, bool>>
+        mapperResponse;
+    auto mapper = bus.new_method_call(
+        "xyz.openbmc_project.User.Manager", "/xyz/openbmc_project/user",
+        "xyz.openbmc_project.User.Manager", "GetUserInfo");
+
+    std::string userName = getSessionItem(sessionId)->getOwner();
+    mapper.append(userName.c_str());
+
+    auto mapperResponseMsg = bus.call(mapper);
+
+    if (mapperResponseMsg.is_method_error())
+    {
+        log<level::ERR>("Error getting user info",
+                        entry("SERVICE=%s", "xyz.openbmc_project.User.Manager"),
+                        entry("USERNAME=%s", "admin"));
```

Comment:
I guess here should be `userName` instead of `"admin"`. Isn't?

Reply:
@nest1ing Fixed + eveyrwhere I found the same issue also fixed so fixed in several places